src/components/core/accordion/accordion.tsx

```diff
@@ -0,0 +1,164 @@
+    onToggle: (expand: boolean) => void;
+};
+
+const AccordionDiv = styled('div', {
+    variants: {
+        elevation_type: {
+            xs: {
+                boxShadow: 'none',
+            },
+            sm: {
+                boxShadow: 'none',
+            },
+            md: {
+                boxShadow: 'none',
+            },
+            lg: {
+                boxShadow: 'none',
+            },
+            xl: {
+                boxShadow: 'none',
+            },
+            xxl: {
+                boxShadow: 'none',
+            },
+            xxxl: {
+                boxShadow: 'none',
+            },
+        },
+        type: {
+            bottomBorder: {
+                borderBottom: '1px solid $greyLight300',
+            },
+            container: {
+                borderRadius: '24px',
+            },
+            containerWithBorder: {
+                borderRadius: '8px',
+                border: '1px solid $greyLight300',
+            },
+            containerWithShadow: {
+                boxShadow: '$xs',
+            },
+        },
+        dark: {
+            true: {
+                borderColor: '$greyDark500',
+            },
+        },
+    },
+    compoundVariants: [
+        {
+            type: 'bottomBorder',
+            dark: true,
+            css: {
+                borderColor: '$greyDark500',
+            },
+        },
+        {
+            type: 'containerWithBorder',
+            dark: true,
+            css: {
+                borderColor: '$greyDark500',
+            },
+        },
+        {
+            type: 'containerWithShadow',
+            elevation_type: 'xs',
+            css: {
+                elevationBoxShadow: 'xs',
+            },
+        },
+        {
+            type: 'containerWithShadow',
+            elevation_type: 'sm',
+            css: {
+                elevationBoxShadow: 'sm',
+            },
+        },
+        {
+            type: 'containerWithShadow',
+            elevation_type: 'md',
+            css: {
+                elevationBoxShadow: 'md',
+            },
+        },
+        {
+            type: 'containerWithShadow',
+            elevation_type: 'lg',
+            css: {
+                elevationBoxShadow: 'lg',
+            },
+        },
+        {
+            type: 'containerWithShadow',
+            elevation_type: 'xl',
+            css: {
+                elevationBoxShadow: 'xl',
+            },
+        },
+        {
+            type: 'containerWithShadow',
+            elevation_type: 'xxl',
+            css: {
+                elevationBoxShadow: 'xxl',
+            },
+        },
+        {
+            type: 'containerWithShadow',
+            elevation_type: 'xxxl',
+            css: {
+                elevationBoxShadow: 'xxxl',
+            },
+        },
+    ],
+});
+
+const Accordion = ({ type, children, elevation_type = 'xs', expand_section, onToggle }: AccordionProps) => {
```

Comment:
```suggestion
const Accordion = ({ type, children, elevation_type = 'xs', expand_section, onToggle }: AccordionProps) => {
```
Check the code suggestions in [**Comment 1**] could apply for here as well